### PR TITLE
Return 404 if disabled format is requested

### LINKED ISSUE #13525: [Bug]: Dont output avif if disabled in thumbnail settings
### Expected behavior

Dont output avif if disabled

### Actual behavior

Avif is included in picture tags for thumbnails, even if they are disabled.

### Steps to reproduce

1. Enable avif via config:
```
pimcore:
    assets:
        image:
            thumbnails:
                auto_formats:
                    webp:
                        enabled: true
                    avif:
                        enabled: true
                        quality: 85
```

2. Generate thumbnails for any image with format "auto"

3. Disable avif via config:
```
pimcore:
    assets:
        image:
            thumbnails:
                auto_formats:
                    webp:
                        enabled: true
                    avif:
                        enabled: false
```

4. Generate output for picture tag, avif format will be included in the html output, even if disabled

### PR DESCRIPTION
**Pull request by aryaantony92**
## Changes in this pull request  
Resolves #13525

### CODE REVIEW

File: bundles/CoreBundle/src/Controller/PublicServicesController.php [hunk 1]
```diff
@@ -49,6 +49,22 @@ public function thumbnailAction(Request $request): RedirectResponse|StreamedResp
...
+        if($asset instanceof Asset\Image) {
+            // Return 404 if the requested thumbnail format is disabled from the config
+            $thumbnailFormats = ThumbnailConfig::getAutoFormats();
+            if(!in_array($requestedFileExtension, ['jpg', 'jpeg'])) {
+                if(empty($thumbnailFormats))
+                    return new Response("HTTP/1.1 404 Not Found\n Requested thumbnail format is disabled", 404);
+
+                foreach($thumbnailFormats as $autoFormat => $autoFormatConfig) {
+                    if(!$autoFormatConfig['enabled'] && $requestedFileExtension == $autoFormat) {
+                        return new Response("HTTP/1.1 404 Not Found\nRequested thumbnail format is disabled from the config", 404);
```
Comment (kingjia90) [inline bundles/CoreBundle/src/Controller/PublicServicesController.php]:
```suggestion
                    return new Response("Requested thumbnail format is disabled", Response::HTTP_NOT_FOUND);

                foreach($thumbnailFormats as $autoFormat => $autoFormatConfig) {
                    if(!$autoFormatConfig['enabled'] && $requestedFileExtension == $autoFormat) {
                        return new Response("Requested thumbnail format is disabled from the config", Response::HTTP_NOT_FOUND);
```

Comment (kingjia90) [reply bundles/CoreBundle/src/Controller/PublicServicesController.php]: or what about using `throw new NotFoundHttpException('Requested thumbnail format is disabled');`?

File: bundles/CoreBundle/src/Controller/PublicServicesController.php [hunk 2]
```diff
@@ -49,6 +49,22 @@ public function thumbnailAction(Request $request): RedirectResponse|StreamedResp
             'file_extension' => $requestedFileExtension,
         ];
 
+        $asset = Asset::getById($config['asset_id']);
+        if($asset instanceof Asset\Image) {
+            // Return 404 if the requested thumbnail format is disabled from the config
+            $thumbnailFormats = ThumbnailConfig::getAutoFormats();
+            if(!in_array($requestedFileExtension, ['jpg', 'jpeg'])) {
```
Comment (kingjia90) [inline bundles/CoreBundle/src/Controller/PublicServicesController.php]: what's the reason behind this exception for jpg file formats?

File: bundles/CoreBundle/src/Controller/PublicServicesController.php [hunk 3]
```diff
@@ -49,6 +49,22 @@ public function thumbnailAction(Request $request): RedirectResponse|StreamedResp
...
+                if(empty($thumbnailFormats))
+                    return new Response("HTTP/1.1 404 Not Found\n Requested thumbnail format is disabled", 404);
+
+                foreach($thumbnailFormats as $autoFormat => $autoFormatConfig) {
+                    if(!$autoFormatConfig['enabled'] && $requestedFileExtension == $autoFormat) {
```
Comment (kingjia90) [inline bundles/CoreBundle/src/Controller/PublicServicesController.php]:
```suggestion
                    if($requestedFileExtension == $autoFormat && !$autoFormatConfig['enabled']) {
```
imho by swapping order, is easier to read in a loop

**Review by kingjia90 — CHANGES_REQUESTED** [review]
If the issue #13525 has `bc break` label and the PR is for 11.x, then would probably need some change in docs 
https://github.com/pimcore/pimcore/blob/b23b847c42a79fce4ed6905570283a04ae2b221f/doc/04_Assets/03_Working_with_Thumbnails/01_Image_Thumbnails.md?plain=1#L437-L440
and a mention in the upgrade notes.

Or is it a non breaking change that should go in 10.5.x?

**Comment by kingjia90** [comment]
Probably the code should throw expection and be moved under https://github.com/pimcore/pimcore/blob/b23b847c42a79fce4ed6905570283a04ae2b221f/models/Asset/Service.php#L579
 (in `Asset\Service::getImageThumbnailByArrayConfig`) so that protected asset would also work that way, see https://github.com/pimcore/pimcore/pull/14155

File: bundles/CoreBundle/src/Controller/PublicServicesController.php (hunk 2 above)
Comment (aryaantony92) [reply bundles/CoreBundle/src/Controller/PublicServicesController.php]: The reason I avoided the check for jpg file formats i that jpg image thumbnails are generated and displayed in the picture tag even if jpg/jpeg is disabled from the config

**Review by kingjia90 — APPROVED** [review]

**Pull request merged**